src/dataflow/components/dataflow-program-topbar.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { ProgramRunTime, ProgramRunTimes } from "../utilities/node";
+
+import "./dataflow-program-topbar.sass";
+
+interface IProps {
+  onRunProgramClick: () => void;
+  onStopProgramClick: () => void;
+  onProgramTimeSelectClick: (type: number) => void;
+  isRunEnabled: boolean;
+}
+
+export class DataflowProgramTopbar extends React.Component<IProps, {}> {
+  constructor(props: IProps) {
+    super(props);
+  }
+
+  public render() {
+    return (
+      <div className="program-editor-topbar">
+        <div>Duration:</div>
+        <select
+          onChange={this.handleSelectChange}
+          disabled={this.props.isRunEnabled}
+        >
+          { ProgramRunTimes.map((rt: ProgramRunTime, i: number) => (
+              <option key={i} value={rt.val}>
+                {rt.text}
+              </option>
+            ))
+          }
+        </select>
+        <button
+          onClick={this.props.onRunProgramClick}
+          disabled={this.props.isRunEnabled}
+        >
+          Run
+        </button>
+        <button
+          onClick={this.props.onStopProgramClick}
+          disabled={!this.props.isRunEnabled}
+        >
+          Stop
+        </button>
+      </div>
+    );
+  }
```

Comment:
If, by chance, you saw the mock-ups with the program record interval in the topbar (used for determining how frequently we write to a dataset), I think we can ignore that for now.  Referring back to previous discussions and DF2, we want the record interval in the Data Storage Node (which is what is already built and what was discussed and agreed upon several times).